lib/inventory/metadata/metadata.go

```diff
@@ -225,25 +221,54 @@ func (c *fetchConfig) fetchContainerOrchestrator() string {
 
 // fetchCloudEnvironment returns aws, gpc or azure if the instance is running on
 // such cloud environments.
-func (c *fetchConfig) fetchCloudEnvironment() string {
-	if c.awsHTTPGetSuccess() {
-		return "aws"
-	}
-	if c.gcpHTTPGetSuccess() {
-		return "gcp"
-	}
-	if c.azureHTTPGetSuccess() {
-		return "azure"
+func (c *fetchConfig) fetchCloudEnvironment(ctx context.Context) string {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
```

Comment:
If we are going to ignore the results after 3 seconds we might as well make this a timeout context, and the check below can be for `<-ctx.Done()`
```suggestion
	ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
	defer cancel()
```